### ISSUE TITLE
testusdviewNavigationKeys fails when tests run multithreaded

### ISSUE THREAD

**Issue by elrond79** [PixarAnimationStudios/USD#341]
### Description of Issue

The testusdviewNavigationKeys test seems to fail when run via ctest in parallel, with a lot of other usdview tests.  My guess is that, when multiple other usdview windows are being opened at the same time, some other usdview window is stealing the keyboard focus.  Potentially this issue could be fixed by somehow ensuring that the keystrokes used in the test are sent to a specific window?
Also, it's worth nothing that these test failures were occurring when running w/ 16 different tests in parallel (on a machine with 16 physical cores).  The tests seem to be stable when run non-parallel, but we'd like to be able to continue using parallel testing, since it speeds up the tests greatly.

### Steps to Reproduce
1. cd to USD build directory
2. ctest -j16 -R 'usdview'

If this fails, try again (on my computer, the second, and all subsequent trials, ran the testusdviewNavigationKeys test FIRST, and it always fails.  Perhaps this has something to do with the failure?)

### System Information (OS, Hardware)
16-core Intel
Linux (CentOS 7)

### Package Versions

Built from USD 0.8.1+, 4f3b44f08a309684e343a908f5d1042292054c2c

**Comment by jtran56** [comment]
Filed as internal issue #154664.

**Comment by spiffmon** [comment]
Hi Paul,
   Yes, I noted this during the development, and in our Scons/mentor test system, this test is identified as needing to run in isolation.  Is there a way to indicate that with ctest?
   The feature is meant to work for all windows belonging to the application, so we send all the key events to the application object.

Thanks!
--spiff

**Comment by elrond79** [comment]
Looks like there is:

https://cmake.org/cmake/help/v2.8.12/cmake.html#prop_test:RUN_SERIAL

**Comment by elrond79** [comment]
I take it then internally you're not using ctest? Are there any other tests that you have marked as non-parallel? Because we've had a handle of other tests that seem to fail occasionally, but not frequently enough for us to be able to conclusively say it's because of parallel testing (or else we just haven't invested the time to find out yet).

**Comment by superfunc** [comment]
Hey @elrond79 we are using ctest on our open source CI setup, however, the VM resources allocated to that CI system are headless so they aren't running the testusdview* suite currently.

**Comment by elrond79** [comment]
Makes sense. Was just hoping that this same thing might magically fix our other tests. =)

Anyway, made a PR for this: https://github.com/PixarAnimationStudios/USD/pull/344

**Comment by spiffmon** [comment]
Thanks, Paul!  Funnily enough, we coded up pretty much that change today, and applied it also to two tests in libwork that we run serially in Scons.  So if your mysterious test failures were in work, then maybe it'll be good!
